cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CruiseControlReconciler.java

```diff
@@ -247,52 +244,51 @@ protected Future<Void> certificatesSecret(Clock clock) {
-                    secretOperator.getAsync(reconciliation.namespace(), KafkaResources.entityTopicOperatorCcApiSecretName(reconciliation.name()))
-                ).compose(
+            Future<Secret> centralizedApiSecretFuture = secretOperator.getAsync(reconciliation.namespace(), CruiseControlResources.apiSecretName(reconciliation.name()));
+            Future<Secret> userManagedApiSecretFuture = cruiseControl.userManagedApiUsersEnabled()
+                    ? secretOperator.getAsync(reconciliation.namespace(), cruiseControl.getUserManagedApiSecretName())
+                    : Future.succeededFuture(null);
+            Future<Secret> topicOperatorManagedApiSecretFuture = isTopicOperatorEnabled
+                    ? secretOperator.getAsync(reconciliation.namespace(), KafkaResources.entityTopicOperatorCcApiSecretName(reconciliation.name()))
+                    : Future.succeededFuture(null);
+            return Future.join(centralizedApiSecretFuture, userManagedApiSecretFuture, topicOperatorManagedApiSecretFuture)
+                .compose(
                     compositeFuture -> {
-                        Secret oldSecret = compositeFuture.resultAt(0);
-                        Secret topicOperatorApiSecret = compositeFuture.resultAt(1);
-                        String cruiseControlAuthFile = oldSecret != null ? Util.decodeFromBase64(oldSecret.getData().get(API_AUTH_FILE_KEY)) : null;
-                        CruiseControl.CruiseControlUser toAdminUser = topicOperatorApiSecret != null
-                            ? new CruiseControl.CruiseControlUser(Util.decodeFromBase64(topicOperatorApiSecret.getData().get(API_TO_ADMIN_NAME_KEY)),
-                            Util.decodeFromBase64(topicOperatorApiSecret.getData().get(API_TO_ADMIN_PASSWORD_KEY)))
-                            : null;
-                        // generate a new CC API secret if there is no CC auth file, or there is no TO API secret, or TO admin password changed
-                        Secret newSecret = cruiseControlAuthFile == null || toAdminUser == null || !cruiseControlAuthFile.contains(toAdminUser.password())
-                            ? cruiseControl.generateApiSecret(passwordGenerator, null, toAdminUser)
-                            : cruiseControl.generateApiSecret(passwordGenerator, oldSecret, null);
-                        this.apiSecretHash = ReconcilerUtils.hashSecretContent(newSecret);
-                        return secretOperator.reconcile(reconciliation, reconciliation.namespace(), CruiseControlResources.apiSecretName(reconciliation.name()), newSecret)
+                        Secret oldCentralizedApiSecret = compositeFuture.resultAt(0);
+                        Secret userManagedApiSecret = compositeFuture.resultAt(1);
+                        Secret topicOperatorManagedApiSecret = compositeFuture.resultAt(2);
+
+                        Secret newCentralizedApiUsersSecret = ApiCredentials.generateApiSecret(cruiseControl, passwordGenerator,
```

Comment:
Same as above, I would call it `ccApiSecret` or something like that. In the model class, you also don't call it `geenrateCentralizedApiSecret` either.